Mathlib/CategoryTheory/Enriched/Ordinary.lean

```diff
@@ -104,6 +113,35 @@ lemma eHomWhiskerLeft_comp (X : C) {Y Y' Y'' : C} (g : Y ⟶ Y') (g' : Y' ⟶ Y'
     associator_inv_naturality_right_assoc, Iso.hom_inv_id_assoc,
     whisker_exchange_assoc, MonoidalCategory.whiskerRight_id_assoc, Iso.inv_hom_id_assoc]
 
+/-- Whiskering commutes with the enriched composition. -/
+@[reassoc]
+lemma eComp_eHomWhiskerLeft (X Y : C) {Z Z' : C} (g : Z ⟶ Z') :
+    eComp V X Y Z ≫ eHomWhiskerLeft V X g =
+      _ ◁ eHomWhiskerLeft V Y g ≫ eComp V X Y Z' := by
+  unfold eHomWhiskerLeft
+  rw [rightUnitor_inv_naturality_assoc, ← whisker_exchange_assoc]
```

Comment:
```suggestion
  dsimp [eHomWhiskerLeft]
  rw [rightUnitor_inv_naturality_assoc, ← whisker_exchange_assoc]
```

Reply:
Is there a reason you prefer `dsimp` to `unfold`? 

In Coq I believe `unfold` would be the "weakest" version of "replace this thing with it's definition", but it seems that is not the case in lean. As I understand it, `unfold` in lean is just `simp` with some special settings, but I haven't been able to figure out what those settings are.

Reply:
I don't think there is much difference between `unfold foo` and `dsimp [foo]`, the suggestion was just to conform with the style of the surrounding proofs. I also think the `dsimp` version is more common in mathlib, but that's just a feeling.

Reply:
That's helpful, thank you.